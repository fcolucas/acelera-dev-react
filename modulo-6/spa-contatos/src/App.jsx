import React from 'react';

import './App.scss';
import Topbar from './components/Topbar';
import Filters from './components/Filters';
import Contacts from './components/Contacts';
import Contact from './components/Contact';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      contacts: [],
      filteredContacts: [],
      sortBy: '',
      loading: false
    }
  }

  componentDidMount() {
    this.setState({ loading: true })

    fetch('https://5e82ac6c78337f00160ae496.mockapi.io/api/v1/contacts')
      .then(response => response.json())
      .then(data => this.setState({ 
        contacts: data,
        filteredContacts: data,
        loading: false
      }));
  }

  filterContacts = textFilter => {
    let filteredContacts = this.state.contacts
    filteredContacts = filteredContacts.filter(contact => {
      return contact.name.toLowerCase().indexOf(
        textFilter.toLowerCase()) !== -1
    })
    this.setState({ filteredContacts })
  }

  sortFilterContacts = value => {
    const compareValues = (key, order = 'asc') => {
      return (a, b) => {
        if(!a.hasOwnProperty(key) || !b.hasOwnProperty(key)) return 0;

        const A = (typeof a[key] === 'string') ? a[key].toLowerCase() : a[key];
        const B = (typeof b[key] === 'string') ? b[key].toLowerCase() : b[key];

        let aux = A > B ? 1 : -1;
        
        return(
          (order === 'desc') ? aux * -1 : aux
        );
      };
    };

    const { filteredContacts, sortBy } = this.state;
    let sortValue, filtered;

    if(value === sortBy) {
      sortValue = '';
      filtered = [...filteredContacts];
    }
    else {
      sortValue = value;
      filtered = filteredContacts.sort(compareValues(value));
    }

    this.setState({
      sortBy: sortValue,
      filteredContacts: filtered
    })
  }

  render() {
    const { filteredContacts, sortBy, loading } = this.state;

    return (
      <React.Fragment>
        <div class="app" data-testid="app">
          <Topbar />

          <Filters 
            onChange={this.filterContacts}
            doSort={this.sortFilterContacts}
            selectFilter={sortBy}
            />

          <Contacts>
            {loading ? <h2>Carregando dados...</h2> : null}
            {filteredContacts.map(contact => 
              <Contact key={contact.id} data={contact}/>
            )}
          </Contacts>
        </div>
      </React.Fragment>
    )
  }
}

export default App;
