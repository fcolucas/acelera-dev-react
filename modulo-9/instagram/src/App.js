import React from 'react';
import './index.css';

import { useSelector, useDispatch } from 'react-redux';

import { setLike } from './actions';

import Post from './components/Post';
import Story from '././components/Story';
import Header from './components/Header';

function App() {
  const { stories, posts } = useSelector( state => state );
  const dispatch = useDispatch();

  const handleLike = (id) => {
    dispatch(setLike(id));
  }

  return (
    <div>
      <Header />
      
      <section className="stories">
        <div className="container">
          <div className="stories__list">
            {stories.map(story => (
              <Story key={story.id} story={story}/>
            ))}
          </div>
        </div>
      </section>

      <main className="main__post">
        <div className="container">
          {posts.map(post => (
            <Post key={post.id} post={post} onClickLike={handleLike}/>
          ))}
        </div>
      </main>
    </div>
  );
}

export default App;
