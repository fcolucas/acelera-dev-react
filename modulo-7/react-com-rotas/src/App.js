import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { Link } from 'react-router-dom';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p className="App-intro">
            <Link to="/sobre">Ir para p√°gina sobre</Link>
            Edit <code>src/App.js</code> and save to reload.
          </p>
        </header>
      </div>
    );
  
  }

}

export default App;
