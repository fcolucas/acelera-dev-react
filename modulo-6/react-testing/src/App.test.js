import React from "React";
import App from "./App";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import Home from "./Home";

it("deve renderizar a página inicial do usuário não logado", () => {
  const props = {
    isLoggedIn: false,
    username: null
  }
  const app = mount(<App {...props} />)
  expect(toJson(app)).toMatchSnapshot()
})

it("deve renderizar a página inicial do usuário logado (Heisenberg)", () => {
  const props = {
    isLoggedIn: true,
    username: "Heisenberg"
  }
  const app = mount(<App {...props} />)
  expect(toJson(app)).toMatchSnapshot()
})

it("deve renderizar a página inicial do usuário sem tarefas", () => {
  const props = {
    isLoggedIn: true,
    username: "Jessie"
  }
  const app = mount(<App {...props} />)
  expect(toJson(app)).toMatchSnapshot()
})

// it("deve renderizar a página inicial do usuário não conectado", () => {
//   const props = {
//     isLoggedIn: false,
//     username: null
//   }
//   const app = mount(<App {...props} />)

//   //Checking output
//   expect(app.find(Home)).toHaveLength(0);
//   expect(app.find('p').at(0).text()).toEqual("Hello, visitor. Sign in to continue.");
// })

// it("deve renderizar a página inicial do usuário logado(Heisenberg)", () => {
//   const props = {
//     isLoggedIn: true,
//     username: "Heisenberg"
//   }
//   const app = mount(<App {...props} />)

//   //Checking output
//   expect(app.find(Home)).toHaveLength(1)
//   expect(app.find('p').at(0).text()).toEqual("Welcome, Heisenberg!")
  
//   expect(app.find('li')).toHaveLength(4)
//   expect(app.find('li').at(0).text()).toEqual("8am - Chemistry classes at school")
//   expect(app.find('li').at(1).text()).toEqual("12:30pm - Meet Jesse for lunch")
//   expect(app.find('li').at(2).text()).toEqual("15pm - Meet Gus at the Pollos Hermanos")
//   expect(app.find('li').at(3).text()).toEqual("20pm - Dinner by the pool with Hank and Marie")
// })

// it("deve renderizar a página inicial para usuário logado sem tarefas", () => {
//   const props = {
//     isLoggedIn: true,
//     username: "Jessie"
//   }
//   const app = mount(<App {...props} />)

//   //Checking output
//   expect(app.find(Home)).toHaveLength(1)
//   expect(app.find('p').at(0).text()).toEqual("Welcome, Jessie!")

//   expect(app.find('li')).toHaveLength(0)
// })