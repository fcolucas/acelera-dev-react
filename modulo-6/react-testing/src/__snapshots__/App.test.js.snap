// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deve renderizar a página inicial do usuário logado (Heisenberg) 1`] = `
<App
  isLoggedIn={true}
  username="Heisenberg"
>
  <div
    className="App"
  >
    <Home
      tasks={
        Array [
          "8am - Chemistry classes at school",
          "12:30pm - Meet Jesse for lunch",
          "15pm - Meet Gus at the Pollos Hermanos",
          "20pm - Dinner by the pool with Hank and Marie",
        ]
      }
      username="Heisenberg"
    >
      <div>
        <p>
          Welcome, 
          Heisenberg
          !
        </p>
        <ul>
          <li
            key="0"
          >
            8am - Chemistry classes at school
          </li>
          <li
            key="1"
          >
            12:30pm - Meet Jesse for lunch
          </li>
          <li
            key="2"
          >
            15pm - Meet Gus at the Pollos Hermanos
          </li>
          <li
            key="3"
          >
            20pm - Dinner by the pool with Hank and Marie
          </li>
        </ul>
      </div>
    </Home>
  </div>
</App>
`;

exports[`deve renderizar a página inicial do usuário não logado 1`] = `
<App
  isLoggedIn={false}
  username={null}
>
  <div
    className="App"
  >
    <p>
      Hello, visitor. Sign in to continue.
    </p>
  </div>
</App>
`;

exports[`deve renderizar a página inicial do usuário sem tarefas 1`] = `
<App
  isLoggedIn={true}
  username="Jessie"
>
  <div
    className="App"
  >
    <Home
      username="Jessie"
    >
      <div>
        <p>
          Welcome, 
          Jessie
          !
        </p>
        <ul />
      </div>
    </Home>
  </div>
</App>
`;
